nacos/src/main/java/com/linecorp/armeria/common/nacos/NacosConfigSetters.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2024 LY Corporation
+
+ * LY Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.common.nacos;
+
+import com.linecorp.armeria.common.annotation.UnstableApi;
+import com.linecorp.armeria.internal.nacos.NacosClientBuilder;
+
+/**
+ * Sets properties for building a Nacos client.
+ */
+@UnstableApi
+public interface NacosConfigSetters {
```

Comment:
optional; What do you think of generalizing this interface and adding the related methods?

e.g.
```java
public interface NacosConfigSetters<SELF extends NacosConfigSetters<SELF>> {
  SELF nacosApiVersion(...)
}
```